### PR TITLE
Support optional installation root

### PR DESCRIPTION
**Pull request by nanonyme**
Part of https://github.com/pypa/flit/issues/570

### CODE REVIEW

**Comment by nanonyme** [comment]
@takluyver any chance getting opinions/reviews about this and the sibling PR https://github.com/pypa/flit/pull/572?

**Comment by takluyver** [comment]
This one I might accept, but not with the name 'prefix', because `sys.prefix` and pip's `--prefix` argument both mean something different. I think the semantics of this match pip's `--root` option or [installer's](https://installer.readthedocs.io/en/stable/cli/installer/) `--destdir`.

Is this still useful to you without #572? Perhaps we could print the target path to stdout so you could do something like:

```shell
INSTALLED_DIR=$(python bootstrap_install.py ...)
python -m compileall $INSTALLED_DIR
```

**Comment by nanonyme** [comment]
I will need to experiment a bit. I don't like the stdout-as-api approach. Perhaps I can just tell it to operate on install-root recursively, it should not contain content that does not need to be have bytecode compiled.

**Comment by nanonyme** [comment]
Yes, this would be useful standalone. it's very minor extra work compileall has to do to traverse entire install-root.

File: flit_core/bootstrap_install.py [hunk 1]
```diff
@@ -37,12 +37,23 @@ def extract_wheel(whl_path, dest):
...
 
     if not args.wheel.name.startswith('flit_core-'):
         sys.exit("Use this script only for flit_core wheels")
-    if not args.installdir.is_dir():
-        sys.exit(f"{args.installdir} is not a directory")
-
-    extract_wheel(args.wheel, args.installdir)
+    if args.install_root:
+        installdir = args.install_root / args.installdir.relative_to("/")
+        installdir.mkdir(parents=True, exist_ok=True)
```
Comment (takluyver) [inline flit_core/bootstrap_install.py:53]: Let's do the mkdir in either case, for consistency. I don't think there was a particular reason why I was checking for an existing directory - maybe I just wanted to ensure it failed with a clear error if the expectations weren't met.

Comment (nanonyme) [reply flit_core/bootstrap_install.py:53]: We have to, the point with install-root is that install-root *does* exist but installdir *does not*. This is essentially a vanilla destination that is captured and packaged further.

Comment (nanonyme) [reply flit_core/bootstrap_install.py:53]: I see. Well, if I can remove the check, then that would maybe work too.

**Comment by takluyver** [comment]
Thanks!

**Pull request merged**